packages/ifpack2/src/Ifpack2_Details_Chebyshev_def.hpp

```diff
@@ -195,8 +195,37 @@ struct LapackHelper{
                            Teuchos::ArrayRCP<typename Teuchos::ScalarTraits<ScalarType>::magnitudeType> offdiag) {
     throw std::runtime_error("LAPACK does not support the scalar type.");
   }
+
 };
 
+template<class V>
+void
+computeInitialGuessForCG (const V& diagonal, V& x) {
+  using device_type = typename V::node_type::device_type;
+  using range_policy = Kokkos::RangePolicy<typename device_type::execution_space>;
+
+  // Initial randomization of the vector
+  x.randomize();
+
+
+  // Zero the stuff that where the diagonal is equal to one.  These are assumed to
+  // correspond to OAZ rows in the matrix.
+  size_t N = x.getLocalLength();
+  auto d_view = diagonal.template getLocalView<device_type>(Tpetra::Access::ReadOnly);
+  auto x_view = x.template getLocalView<device_type>(Tpetra::Access::ReadWrite);
+
+  auto ONE  = Teuchos::ScalarTraits<typename V::scalar_type>::one();
+  auto ZERO = Teuchos::ScalarTraits<typename V::scalar_type>::zero();
```

Comment:
I think you might need the impl scalar type here. Then again, we probably don't want to call Cheby for complex.